controllers/volumereplicationgroup_controller.go

```diff
@@ -466,15 +466,24 @@ func (v *VRGInstance) restorePVs() error {
 	success := false
 
 	for _, s3ProfileName := range v.instance.Spec.S3ProfileList {
-		pvList, err := v.listPVsFromS3Store(s3ProfileName)
+		pvList, err := v.fetchPVClusterDataFromS3Store(s3ProfileName)
 		if err != nil {
-			v.log.Error(err, "failed to retrieve PVs from S3 store", "ProfileName", s3ProfileName)
+			v.log.Error(err, fmt.Sprintf("error fetching PV cluster data from S3 profile %s", s3ProfileName))
 
 			continue
 		}
 
 		v.log.Info(fmt.Sprintf("Found %d PVs", len(pvList)))
 
+		err = v.sanityCheckPVClusterData(pvList)
+		if err != nil {
+			errMsg := fmt.Sprintf("error found during sanity check of PV cluster data in S3 store %s", s3ProfileName)
+			v.log.Error(err, errMsg)
+			v.log.Error(err, "Resolve the conflict in the above S3 store to deploy the application")
```

Comment:
Maybe you wanted this to be at Info level, otherwise, you'll end up with two stack traces.

Reply:
Fixed it.